### PR TITLE
refactor: 6.0 mparticle

### PR DESCRIPTION
**Pull request by PoornimaApptentive**
Tested the all the interactions except rating dialog. Tested the Person data and Custom Person data.

Code change: Replaced the AttributeListener with `UserAttributeLIstener` & `IdentityListener`. Updating the `mParticle` Id on `Apptentive` registration success.

### CODE REVIEW

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt [hunk 1]
```diff
@@ -41,88 +41,100 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
...
+                    lastKnownLastName = value.toString()
+                }
+                else -> {
+                    addCustomPersonData(key, value.toString())
+                    return
+                }
             }
-            fullName += lastKnownLastName
+            val fullName = listOfNotNull(lastKnownLastName, lastKnownFirstName).joinToString(separator = " ").trim()
+            if (fullName.isNotBlank()) Apptentive.setPersonName(fullName)
```
Comment (ChaseApptentive) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:101]: @frankus does iOS also set set this as Last First or is this possibly a bug?

Comment (frankus) [reply src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:101]: We set use an OS feature called [https://developer.apple.com/documentation/foundation/personnamecomponentsformatter](PersonNameComponentsFormatter) that takes the device locale into account but will be "First Last" for the Anglosphere.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt [hunk 2]
```diff
@@ -41,88 +41,100 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
...
+    override fun onSetUserTag(key: String?, user: FilteredMParticleUser?) {
+       //Ignored
+    }
+
+    override fun onSetUserAttributeList(
+        attributeKey: String?,
+        attributeValueList: MutableList<String>?,
+        user: FilteredMParticleUser?
+    ) {
+        //Ignored
```
Comment (frankus) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:114]: Ignored because supportsAttributeList returns false, right?

**Review by frankus — COMMENTED** [review]
Looks good, but I think the name order should probably be First Last.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt (hunk 2 above)
Comment (PoornimaApptentive) [reply src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:114]: Yes, It was false and ignored in the legacy.

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt [hunk 3]
```diff
@@ -41,88 +42,99 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
...
         if (apptentiveAppKey != null && apptentiveAppSignature != null) {
             val configuration = ApptentiveConfiguration(apptentiveAppKey, apptentiveAppSignature)
-            Apptentive.register(context.applicationContext as Application, configuration)
-            ApptentiveNotificationCenter.defaultCenter()
-                .addObserver(
-                    ApptentiveNotifications.NOTIFICATION_CONVERSATION_STATE_DID_CHANGE,
-                    this
-                )
+            configuration.logLevel = getApptentiveLogLevel()
+            configuration.distributionVersion = com.mparticle.BuildConfig.VERSION_NAME
```
Comment (PoornimaApptentive) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:46]: Verified with the mP team about using their BuildConfig.VERSION_NAME for distributionVersion

File: src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt [hunk 4]
```diff
@@ -159,134 +194,83 @@ class ApptentiveKit : KitIntegration(), KitIntegration.EventListener, CommerceLi
...
+        //Ignored
     }
 
     //endregion
+
     //region Helpers
-    private fun engage(context: Context, event: String, customData: Map<String, String>?) {
-        engage(context, event, customData, *arrayOf())
+    private fun getApptentiveLogLevel(): LogLevel {
+        return when (Logger.getMinLogLevel()) {
```
Comment (PoornimaApptentive) [inline src/main/kotlin/com/mparticle/kits/ApptentiveKit.kt:236]: order the loglevel low to high

**Pull request merged**